modules/gdscript/gdscript_parser.cpp

```diff
@@ -2025,6 +2025,16 @@ GDScriptParser::ExpressionNode *GDScriptParser::parse_unary_operator(ExpressionN
 	return operation;
 }
 
+GDScriptParser::ExpressionNode *GDScriptParser::parse_binary_not_in_operator(ExpressionNode *p_previous_operand, bool p_can_assign) {
+	// check that the NOT if followed by IN by consuming IN,
+	// but leave NOT in previous for the call to parse_binary_operator
+	// which can then treat it as shorthand for the "not in" infix operator:
+	GDScriptTokenizer::Token op = previous;
+	consume(GDScriptTokenizer::Token::IN, R"(Expected "in" after "not" in content-test operator.)");
+	previous = op;
```

Comment:
I don't think it can be done inside the switch case - we have to consume the IN before parsing the right operand and that means checking for, and special-casing, `op == NOT` in every call to `parse_binary_operator`. Not a huge hit, performance-wise, but still easily avoidable by just adding this new function and switching to it in the prefix/infix table that will be checked anyway. It's the only operator that consists of two tokens, the new function seems to be a clean way to hide that from `parse_binary_operator`.

Changing `previous` was necessary since that indicates the type of operator inside `parse_binary_operator` and it's only one token, so NOT had to get there to differentiate it from a plain IN. But with the previous comment re: expanding NOT IN into to two operations instead of a new Variant operator, manipulating `previous` won't be necessary any more, so I'll change that.

However, I still think the new function is better than having an `if` in every call to `parse_binary_operator`. What do you think @vnen ?

Reply:
One more note after updating the patch: using a separate function would also make it very easy to add other two-token operators such as "is not", the other part of godotengine/godot-proposals#1319. I would then rename the function to `parse_two_token_operator` for example, add it as the infix function for `is` as well, and have a simple switch or if/else that checks that `not` is followed by `is` or that `is` is optionally followed by `not`.